contracts/market/template/MarketStorage.sol

```diff
@@ -103,7 +110,7 @@ contract MarketStorage is IMarketCommon {
   address internal immutable paymentToken;
 
   constructor(address _paymentToken, IRegistry _registry) {
-    if (_paymentToken == address(0)) revert InvalidAddress({invalidAddress: _paymentToken});
+    require(IERC20Metadata(_paymentToken).decimals() == 18, "wrong decimals");
```

Comment:
A slightly stronger check than before.

The system still works with payment tokens of other decimals - but some of the checks on initialization would need to be updated. So for usdc this would need to be 6 and the minimums for market seed and mint would also need to be increased (here are those minimums: https://github.com/sherlock-audit/2022-11-float-capital/blob/090c1096aacc0e7dc31bc1d00a82357f9c76fbd4/contracts/market/template/MarketExtended.sol#L125 and https://github.com/sherlock-audit/2022-11-float-capital/blob/090c1096aacc0e7dc31bc1d00a82357f9c76fbd4/contracts/market/template/MarketCore.sol#L265) Nothing else in the code needs to change to work with other decimals AFAIK.